ddtrace/propagation/http.py

```diff
@@ -108,6 +120,35 @@ def _inject(span_context, headers):
         if span_context.dd_origin is not None:
             headers[HTTP_HEADER_ORIGIN] = ensure_text(span_context.dd_origin)
 
+        # Do not try to encode tags if we have already tried and received an error
+        if "_dd.propagation_error" in span_context._meta:
+            return
+
+        # TODO: add whitelist
```

Comment:
```suggestion
        # TODO: add allowlist
```

Reply:
👍🏻 we definitely want this

Reply:
we also need to check if this feature is enabled

Reply:
@brettlangdon what "feature" do you mean here? It would seem that in this PR we should only support an allowed list of tags, specifically just `_dd.p.dm` and `_dd.propagation.error`.

Reply:
propagating tags via headers.

```
DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH: default=128

If the value is 0, we can consider the propagating feature is disabled. We still should parse the incoming x-datadog-tags header even if the propagation was disabled.

Trace tags propagation is disabled: set _dd.propagation_error:disabled  in the root span
```